src/hooks/useCart/useCart.test.tsx

```diff
@@ -1332,6 +1333,144 @@ describe("useCart", () => {
       ]);
     });
 
+    it("should set a 400 cartError when there is an ErrorWithCode with status code 400", async () => {
+      expect.assertions(3);
+      const ids = ["ID1"];
+      const { result } = renderHook(
+        () => useCart(ids, key, endpoint, mockQuotaResSingleId.remainingQuota),
+        {
+          wrapper,
+        }
+      );
+
+      await waitFor(() => {
+        result.current.updateCart("toilet-paper", 2, [
+          {
+            label: "first",
+            value: "first",
+            textInputType: "STRING",
+          },
+        ]);
+        result.current.updateCart("chocolate", 5, [
+          {
+            label: "last",
+            value: "last",
+            textInputType: "STRING",
+          },
+        ]);
+      });
+
+      mockPostTransaction.mockRejectedValueOnce(
+        new ErrorWithCodes("user error", 400, "ABC")
+      );
+
+      await waitFor(() => {
+        result.current.checkoutCart();
+      });
+
+      expect(result.current.cartError?.message).toBe("user error");
```

Comment:
I'll like to test beyond this LOL, so that I can test that we
- theres no title dialog in the alert
- we show the error code

But I'm not really sure if this is the right place to do it, or what field to call for. Need help for this haha

Reply:
Not sure if we can access the `AlertModalContextProvider` and its functions, such as `showErrorAlert` and check if they are called with the right fields.